react-context/client/app.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import AppContext from './lib/app-context';
+import parseRoute from './lib/parse-route';
+import decodeToken from './lib/decode-token';
+import Auth from './pages/auth';
+import Home from './pages/home';
+import NotFound from './pages/not-found';
+import Navbar from './components/navbar';
+import PageContainer from './components/page-container';
+
+Navbar.contextType = AppContext;
+Auth.contextType = AppContext;
+Home.contextType = AppContext;
```

Comment:
It's better to import the app context into each of the modules that need it and do the assignment there. That way when you read those files, you can see what context the component relies on.